### PR TITLE
[test] Add Eiger to the "Hello, World" tests

### PR DESCRIPTION
**Pull request by ekouts**

### CODE REVIEW

File: cscs-checks/prgenv/helloworld.py [hunk 1]
```diff
@@ -109,7 +109,8 @@ def compile_timer_end(self):
 class HelloWorldTestSerial(HelloWorldBaseTest):
     def __init__(self, lang, linkage):
         super().__init__('serial', lang, linkage)
-        self.valid_systems += ['arolla:pn', 'tsa:pn']
+        if linkage == 'dynamic':
+            self.valid_systems.append('eiger:mc')
```
Comment (vkarak) [inline cscs-checks/prgenv/helloworld.py:113]: Leave a blank line after this one.

File: cscs-checks/prgenv/helloworld.py [hunk 2]
```diff
@@ -109,7 +109,8 @@ def compile_timer_end(self):
 class HelloWorldTestSerial(HelloWorldBaseTest):
     def __init__(self, lang, linkage):
         super().__init__('serial', lang, linkage)
-        self.valid_systems += ['arolla:pn', 'tsa:pn']
```
Comment (vkarak) [inline cscs-checks/prgenv/helloworld.py:112]: Why did you remove this?

**Review by vkarak — CHANGES_REQUESTED** [review]

File: cscs-checks/prgenv/helloworld.py (hunk 2 above)
Comment (ekouts) [reply cscs-checks/prgenv/helloworld.py:112]: Because these systems are already in the valid systems of `HelloWorldBaseTest`.

**Review by vkarak — APPROVED** [review]
